src/transformStateWithClones.js

```diff
@@ -8,6 +8,33 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const arr = [];
+  let copyOfState = { ...state };
+
+  for (const values of actions) {
+    if (values.type === 'addProperties') {
+      Object.assign(copyOfState, values.extraData);
+      arr.push({ ...copyOfState });
+    }
+
+    if (values.type === 'removeProperties') {
+      for (const keysToRemoveValues of values.keysToRemove) {
+        for (const copyOfStateValues in copyOfState) {
+          if (keysToRemoveValues === copyOfStateValues) {
+            delete copyOfState[copyOfStateValues];
+          }
+        }
+      }
+      arr.push({ ...copyOfState });
```

Comment:
Actually you don't need to push the object in array every if condition. You can do it only once after all the conditions